saleor/webhook/observability/worker.py

```diff
@@ -0,0 +1,190 @@
+import atexit
+import logging
+import os
+from queue import Empty, Full, Queue
+from threading import Lock, Thread
+from time import monotonic
+from typing import Any, Callable, Dict, List, Optional
+
+from django.conf import settings
+
+from .buffers import get_buffer
+from .tracing import opentracing_trace
+
+_worker: Optional["BackgroundWorker"] = None
+_TERMINATOR = object()
+FLUSH_TIMEOUT = 2.0
+
+logger = logging.getLogger(__name__)
+
+
+def shutdown(timeout=FLUSH_TIMEOUT) -> bool:
+    global _worker
+    if _worker:
+        logger.debug("[Observability] Background worker shutdown")
+        success = _worker.flush(timeout)
+        _worker = None
+        return success
+    return False
+
+
+def init():
+    global _worker
+    if _worker is None:
+        _worker = BackgroundWorker(
+            batch_size=settings.OBSERVABILITY_BUFFER_BATCH_SIZE,
+            timeout=settings.OBSERVABILITY_REPORT_PERIOD.total_seconds(),
+        )
+        atexit.register(shutdown)
+
+
+def put_event(buffer_name: str, generate_payload: Callable[[], Any]) -> bool:
+    if _worker:
+        return _worker.submit_event(buffer_name, generate_payload)
+    logger.warning("[Observability] Worker not initialized, event dropped")
```

Comment:
I would say putting `[Observability]` in each log is unnecessary, we can already see the module name in JSON logs. But it doesn't hurt to make it clear, so I have mixed feelings about it

Reply:
Please don't prefix log messages. If you want those for local debugging, use a custom format for the console logger.

In production, Saleor outputs logs in JSON format. Maybe you want to add something to `extra`, like the type of event that was dropped?

Reply:
Sorry, I automatically copied the convention from webhooks plugin tasks `"[Webhook] Sending payload to %r for event %r."`. I will fix it and use `extra` while logging, thanks!